File: GatewayUI/react-ui/src/app/models/descriptorType.ts
export interface DescriptorType {
	id: number;
    name: string;
    description: string;
    imageUrl: string;
    label: string;
    tableLabel: string;
	category: string;
    subCategory: string;
    unit: string;
    dataType: string;
    publication: string;
}

export interface GroupedDescriptorTypes {
    [key: string]: DescriptorType[];
}

export interface DisplayNameMap {
    [key: string]: string | undefined;
}

export const displayNameMap: DisplayNameMap = {
    Global: "GLOBAL",
    Pore: "PORE",
    Other: "OTHER",
    // Add other mappings as needed
};

File: GatewayBackend/Data/Models/DescriptorType.cs

namespace Data.Models
{
    public class DescriptorType
	{
		public int Id { get; set; }
		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
		public string Name { get; set; } = null!;
		public string Label { get; set; } = null!;
		public string TableLabel { get; set; } = null!;
		public string Category { get; set; } = null!;
		public string? SubCategory { get; set; }
		public string? Comments { get; set; }
		public string? Unit { get; set; }
		public string DataType { get; set; } = null!;
		public string? ImageUrl { get; set; }
		public string? Description { get; set; }
		public int? PublicationId { get; set; } 
		public Publication? Publication { get; set; }
		public virtual ICollection<GlobalDescriptor> GlobalDescriptors { get; set; } = new List<GlobalDescriptor>();
    	public virtual ICollection<PoreDescriptor> PoreDescriptors { get; set; } = new List<PoreDescriptor>();
   	 	public virtual ICollection<OtherDescriptor> OtherDescriptors { get; set; } = new List<OtherDescriptor>();
		public virtual ICollection<DescriptorTypeDownload> DescriptorTypeDownloads { get; set; } = new List<DescriptorTypeDownload>();
		public virtual ICollection<PublicationDatasetDescriptorRule> PublicationDatasetDescriptorRules { get; set; } = [];

	}
}
